### ISSUE TITLE
Development bottleneck in bringing new versions of Co-Design to QPixel

### ISSUE THREAD

**Issue by trichoplax** [codidact/qpixel#975]
## Problem
Fixes and features introduced in [Co-Design](https://github.com/codidact/co-design) take a long time to show up here in QPixel, so their benefit to Codidact and other QPixel instances is delayed.

## Example
#923 is a high priority bug that prevents users from clicking on various buttons in Codidact unless they know the workaround. The workaround is difficult to find because navigation of the site is restricted by not being able to press buttons. This is particularly restrictive on mobile even for people who know the workaround, and still highly restrictive on desktop for anyone unfamiliar with the workaround.

Despite the bug making it difficult for people to find their way to Meta to raise it, it has been independently raised several times:
- [Notifications and Communities buttons not responding to icon clicks](https://meta.codidact.com/posts/287466)
- [Clicking on the square with an arrow no longer displays community drop-down](https://meta.codidact.com/posts/287503)
- [Can't click notification icon](https://meta.codidact.com/posts/287650)

A [fix was merged into the `develop` branch of Co-Design 8 weeks ago](https://github.com/codidact/co-design/pull/73), and a release to production of the QPixel codebase has been done since then. However, because Co-Design has not had a release, the bug is still present in production despite the fix.

In addition to this example bug being raised several times on Meta, this bottleneck has also been raised recently on Discord:
- [by me](https://discord.com/channels/634104110131445811/668890340199235613/1063907336084783217)
- [by @cellio](https://discord.com/channels/634104110131445811/676843271762214932/1065461519359410187)

## Purpose
There are frequent improvements to both Co-Design and QPixel. There is plenty of developer activity. This process bottleneck is preventing some of that activity from benefitting users.

This bug is not being raised to get the mentioned delayed example into production - there is already a high priority issue open for that. This bug is being raised to assess what can be improved about the general process of getting changes in Co-Design released and QPixel updated to use the latest Co-Design version number.

$\color{red}\text{In addition to bug fixes and features, the}$
$\color{red}\text{bottleneck also delays the inclusion into}$
$\color{red}\text{QPixel of security updates to Co-Design.}$

## My current understanding
- Releasing Co-Design requires someone with understanding of the process and access to the relevant npm account to follow the steps in the [Co-Design releasing guide](https://github.com/codidact/co-design/blob/develop/docs_src/setup/releasing.md)
- Updating QPixel to use the latest Co-Design release requires updating the version number of Co-Design in 4 different places in QPixel

Please correct any misunderstanding - I have no familiarity with these processes.

## Potential improvements
Please suggest any potential improvements. For example:
- Does the release of Co-Design require the same person to produce the release tag on GitHub and to push it to npm? Would splitting those tasks allow taking up less of the time of the limited number of people with npm account access?
- Do we need more people with npm account access? What steps would need to be taken to approve someone? Are there any volunteers who would like to be considered?
- Can the updating of QPixel to show the latest Co-Design version number be centralised so only one place needs to be changed each time? (Assuming this is not already automated in some way)
- QPixel currently needs to be updated to show a new version number every time Co-Design has a release. Does this need to be the case? Since both repositories are controlled by the same organisation, would it be acceptable for QPixel to trust Co-Design and default to using the latest version rather than having a version number hardcoded? (I guess this depends on it being possible to use "latest" rather than a specific version number in the CDN, and whether our current CDN provider allows overriding the content of this each release)

**Comment by cellio** [comment]
This is an important issue.  I don't know what the best approach is, but I hope others reading this will have suggestions.

**Comment by trichoplax** [comment]
Would it help to have a priority label?

**Comment by MoshiKoi** [comment]
> QPixel currently needs to be updated to show a new version number every time Co-Design has a release. Does this need to be the case? Since both repositories are controlled by the same organisation, would it be acceptable for QPixel to trust Co-Design and default to using the latest version rather than having a version number hardcoded? (I guess this depends on it being possible to use "latest" rather than a specific version number in the CDN, and whether our current CDN provider allows overriding the content of this each release)

Linking #1034 here in regards to this.

**Comment by trichoplax** [comment]
If #1034 is fixed, will that be sufficient to close this issue at that point, since @cellio now has access to make new releases of Co-Design? Is there anything else that needs to be done first? Is there desire for more people to have access to release Co-Design to cover future holiday and illness, or is it preferred to keep the number of people with access minimal for security?

**Comment by cellio** [comment]
I now have `npm` access and documentation.  Switching qpixel to use `@latest` would fix that side of it.  I think with all that, it's probably ok if we don't add more people to the access list right now.  The new potential block will now be at PR reviews/merges for co-design, so we should enable qpixel-reviewers here too.

**Comment by trichoplax** [comment]
The points from the issue description are addressed:


## Potential improvements
- **No longer seems necessary as there is an extra person with access.** ~Does the release of Co-Design require the same person to produce the release tag on GitHub and to push it to npm? Would splitting those tasks allow taking up less of the time of the limited number of people with npm account access?~
- **Covered as @cellio now has access.** ~Do we need more people with npm account access? What steps would need to be taken to approve someone? Are there any volunteers who would like to be considered?~
- **No longer relevant since the version number is no longer hardcoded in QPixel.** ~Can the updating of QPixel to show the latest Co-Design version number be centralised so only one place needs to be changed each time? (Assuming this is not already automated in some way)~
- **QPixel now uses `@latest` following the merge of #1035.** ~QPixel currently needs to be updated to show a new version number every time Co-Design has a release. Does this need to be the case? Since both repositories are controlled by the same organisation, would it be acceptable for QPixel to trust Co-Design and default to using the latest version rather than having a version number hardcoded? (I guess this depends on it being possible to use "latest" rather than a specific version number in the CDN, and whether our current CDN provider allows overriding the content of this each release)~
